test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing live iso [live-iso]", func() {
+	liveIsoTest()
+})
+
+func liveIsoTest() {
+	var (
+		liveIsoImageURL = e2eConfig.GetVariable("LIVE_ISO_IMAGE")
+	)
+	BeforeEach(func() {
+		validateGlobals(specName)
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+	It("Should update the bmh with live iso", func() {
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+
+		bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+		Expect(err).NotTo(HaveOccurred(), "Error getting bmhs")
+
+		isoBmh := bmhs[0]
+		Expect(isoBmh).ToNot(BeNil())
+
+		isoBmh.Spec.Image = &bmov1alpha1.Image{
+			URL:          liveIsoImageURL,
+			Checksum:     "",
+			ChecksumType: "",
+			DiskFormat:   pointer.StringPtr("live-iso"),
+		}
+
+		Expect(bootstrapClient.Update(ctx, &isoBmh)).NotTo(HaveOccurred())
+
+		By("waiting for live iso host to be in provisioned state")
+		Eventually(func(g Gomega) {
+			g.Expect(bootstrapClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: bmhs[0].Name}, &isoBmh)).To(Succeed())
+			g.Expect(isoBmh.Status.Provisioning.State).To(Equal(bmov1alpha1.StateProvisioned))
+		}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
```

Comment:
I would suggest to use: `wait-object-provisioned`
```suggestion
		}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(Succeed())
```